kotlin/navigate-route-with-rerouting/src/main/java/com/esri/arcgisruntime/sample/navigateroutewithrerouting/MainActivity.kt

```diff
@@ -0,0 +1,436 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // authentication with an API key or named user
+        // is required to access basemaps and other location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // set the basemap style to be displayed in the layout's MapView
+        // and create the graphic overlay
+        mapView.apply {
+            map = ArcGISMap(BasemapStyle.ARCGIS_NAVIGATION)
+            graphicsOverlays.add(GraphicsOverlay())
+        }
+
+        navigateButton.setOnClickListener {
+            // start the route navigation on button tap
+            startNavigation()
+        }
+        recenterButton.setOnClickListener {
+            // change the mapview to use navigation mode on button tap
+            mapView.locationDisplay.autoPanMode = LocationDisplay.AutoPanMode.NAVIGATION
+        }
+        resetButton.setOnClickListener {
+            mapView.apply {
+                // stop the location data source
+                locationDisplay.locationDataSource.stopAsync().addDoneListener {
+                    // on completed, set up map to display the route overview
+                    setUpMapRoute()
+                }
+                // clear the map's graphics and create a new one
+                graphicsOverlays.clear()
+                graphicsOverlays.add(GraphicsOverlay())
+            }
+
+            // disable the buttons
+            navigateButton.isEnabled = false
+            recenterButton.isEnabled = false
+        }
+
+        // calculate the route from the starting location and destination
+        // and display the route overview
+        setUpMapRoute()
+    }
+
+    /**
+     * Calculates the route from San Diego Convention Center to the Fleet Science Center.
+     * Then sets up the route graphics and displays the route overview.
+     */
+    private fun setUpMapRoute() {
+        // display the route overview status message
+        statusMessageTV.text = getString(R.string.route_overview)
+
+        // set up Android text to speech engine
+        textToSpeech = TextToSpeech(this) {
+            textToSpeech?.language = Locale.ENGLISH
+        }
+
+        // create a route task using the San Diego geodatabase file
+        routeTask = RouteTask(
+            this,
+            getExternalFilesDir(null)?.path + "/sandiego.geodatabase",
+            "Streets_ND"
+        )
+
+        // load the route task
+        routeTask.loadAsync()
+        routeTask.addDoneLoadingListener {
+            if(routeTask.loadStatus == LoadStatus.LOADED){
```

Comment:
```suggestion
            if (routeTask.loadStatus == LoadStatus.LOADED){
```

Just need a Code -> Reformat code run I think